lib/index.js

```diff
@@ -359,80 +217,110 @@ exports.createFunction = async (ctx, functionSpecification) => {
             try {
-                await exports.deleteFunction(ctx);
+                await exports.deleteFunction(ctx, accessToken);
             } catch (_) {}
         }
         throw e;
     }
 };
 
-exports.deleteFunction = async (ctx, boundaryId, functionId) => {
+exports.deleteFunction = async (ctx, accessToken, boundaryId, functionId) => {
     await Superagent.delete(
         `${ctx.body.baseUrl}/v1/account/${ctx.body.accountId}/subscription/${ctx.body.subscriptionId}/boundary/${
             boundaryId || ctx.body.boundaryId
         }/function/${functionId || ctx.body.functionId}`
-    ).set('Authorization', ctx.headers['authorization']); // pass-through authorization
+    ).set('Authorization', `Bearer ${accessToken}`)
+    .ok(res => res.status === 204 || res.status === 404);
 };
 
-exports.getFunctionDefinition = async (ctx, boundaryId, functionId) => {
-    let response = await Superagent.get(
+exports.getFunctionDefinition = async (ctx, accessToken, boundaryId, functionId) => {
+    const response = await Superagent.get(
         `${ctx.body.baseUrl}/v1/account/${ctx.body.accountId}/subscription/${ctx.body.subscriptionId}/boundary/${
             boundaryId || ctx.body.boundaryId
         }/function/${functionId || ctx.body.functionId}`
-    ).set('Authorization', ctx.headers['authorization']); // pass-through authorization
+    ).set('Authorization', `Bearer ${accessToken}`);
 
     return response.body;
 };
 
-exports.getFunctionUrl = async (ctx, boundaryId, functionId) => {
-    let response = await Superagent.get(
+exports.getFunctionUrl = async (ctx, accessToken, boundaryId, functionId) => {
+    const response = await Superagent.get(
         `${ctx.body.baseUrl}/v1/account/${ctx.body.accountId}/subscription/${ctx.body.subscriptionId}/boundary/${
             boundaryId || ctx.body.boundaryId
         }/function/${functionId || ctx.body.functionId}/location`
-    ).set('Authorization', ctx.headers['authorization']); // pass-through authorization
+    ).set('Authorization', `Bearer ${accessToken}`);
 
     return response.body.location;
 };
 
-exports.getStorageClient = (ctx) => {
-    const expiry = 60 * 16; // 15+1 min to align with Lambda lifecycle plus some buffer
-    const accessToken = Jwt.sign({}, Buffer.from(ctx.configuration.fusebit_storage_key, 'base64').toString('utf8'), {
-        algorithm: 'RS256',
-        expiresIn: expiry,
-        audience: ctx.configuration.fusebit_storage_audience,
-        issuer: ctx.configuration.fusebit_storage_issuer_id,
-        subject: ctx.configuration.fusebit_storage_subject,
-        keyid: ctx.configuration.fusebit_storage_key_id,
-        header: { jwtId: Date.now().toString() },
-    });
-
-    const url = generateStorageUrl(ctx.configuration);
+const removeLeadingSlash = (s) => s.replace(/^\/(.+)$/, '$1');
+const removeTrailingSlash = (s) => s.replace(/^(.+)\/$/, '$1');
+
+exports.createStorageClient = async (ctx, accessToken, storageIdPrefix) => {
```

Comment:
Does not need to be async from what I see

Reply:
Correct, it doesn't now. But it a) gives us consistency - you don't need to remember which functions to await and which not in this SDK, b) allows us to actually do something async in this method in the future without it being a breaking change.

Reply:
Feels weird, but okay... no strong opinion.